typing/typemode.ml

```diff
@@ -95,24 +98,37 @@ module Transled_modifiers =
 
 let transl_modifier_annots annots =
   let step modifiers_so_far annot =
-    let { txt = Any_axis_pair (type a) ((axis, mode) : a Axis.t * a); loc } =
+    match
       transl_annot ~annot_type:Modifier ~required_mode_maturity:None
       @@ unpack_mode_annot annot
-    in
-    let (module A) = Axis.get axis in
-    let is_top = A.le A.max mode in
-    if is_top
-    then
-      (* CR layouts v2.8: This warning is disabled for now because transl_type_decl
-         results in 3 calls to transl_annots per user-written annotation. This results
-         in the warning being reported 3 times. *)
-      (* Location.prerr_warning new_raw.loc (Warnings.Mod_by_top new_raw.txt) *)
-      ();
-    let is_dup =
-      Option.is_some (Transled_modifiers.get ~axis modifiers_so_far)
-    in
-    if is_dup then raise (Error (annot.loc, Duplicated_axis axis));
-    Transled_modifiers.set ~axis modifiers_so_far (Some { txt = mode; loc })
+    with
+    | { txt = Any_axis_pair (type a) ((axis, mode) : a Axis.t * a); loc } ->
+      let (module A) = Axis.get axis in
+      let is_top = A.le A.max mode in
+      if is_top
+      then
+        (* CR layouts v2.8: This warning is disabled for now because transl_type_decl
+           results in 3 calls to transl_annots per user-written annotation. This results
+           in the warning being reported 3 times. *)
+        (* Location.prerr_warning new_raw.loc (Warnings.Mod_by_top new_raw.txt) *)
+        ();
+      let is_dup =
+        Option.is_some (Transled_modifiers.get ~axis modifiers_so_far)
+      in
+      if is_dup then raise (Error (annot.loc, Duplicated_axis axis));
+      Transled_modifiers.set ~axis modifiers_so_far (Some { txt = mode; loc })
+    | { txt = Everything_but_nullability; loc } ->
+      Transled_modifiers.
+        { locality = Some { txt = Locality.Const.min; loc };
+          linearity = Some { txt = Linearity.Const.min; loc };
+          uniqueness = Some { txt = Uniqueness.Const.max; loc };
+          portability = Some { txt = Portability.Const.min; loc };
+          contention = Some { txt = Contention.Const.max; loc };
+          yielding = Some { txt = Yielding.Const.min; loc };
+          externality = Some { txt = Externality.min; loc };
+          nullability =
+            Transled_modifiers.get ~axis:(Nonmodal Nullability) modifiers_so_far
+        }
   in
```

Comment:
this seems worth floating out to a top-level declaration? maybe `Transled_modifiers.everything`?

Reply:
I spent a few minutes on this, but have been defeated by the quantity of abstraction between here and there.  In this module, I can use the concrete options I've written here.

It looks like `Transled_modfiers` is an application of `Jkind_axis.Axis_Collection.Indexed`, which is itself an application of another functor `Indexed_gen`, and in the body of `Indexed_gen` the type of the individual fields of this record are not `const loc option`s but instead are `T.t` where `T` is a parameter whose type is the helpfully named `Misc.T2`, and taht signature defines `t` as abstract.  So, how do I actually build one of these over in that functor?  Also, my head hurts.

Perhaps this is clear to a with-kinds expert - feel free to modify this code and push.

Reply:
I went ahead and did this. But actually I think it's better not to: we can't pre-allocated the `everything` node because it contains the location. And I think it's a bit cleaner just to have the definition of everything right where it is. The one change I would make is to use `Const_op.min` for the past axes. (At some point, `Const_op.t` and `Const.t` will become not equal, forcing this change anyway.) I'll push this change.

Reply:
@goldfirere you mentioned pushing a change here.  If you still have that around, could you push it?  I think this is the last piece missing for this PR and it would be good to get it in the next roll.